service/eventlog/internal/src/main/java/org/eclipse/kapua/service/eventlog/internal/EventLogServiceImpl.java

```diff
@@ -0,0 +1,194 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eurotech and/or its affiliates and others
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.kapua.service.eventlog.internal;
+
+import org.eclipse.kapua.KapuaEntityNotFoundException;
+import org.eclipse.kapua.KapuaException;
+import org.eclipse.kapua.commons.configuration.AbstractKapuaConfigurableService;
+import org.eclipse.kapua.commons.security.KapuaSecurityUtils;
+import org.eclipse.kapua.commons.setting.system.SystemSetting;
+import org.eclipse.kapua.commons.setting.system.SystemSettingKey;
+import org.eclipse.kapua.commons.util.ArgumentValidator;
+import org.eclipse.kapua.event.ListenServiceEvent;
+import org.eclipse.kapua.event.ServiceEvent;
+import org.eclipse.kapua.locator.KapuaLocator;
+import org.eclipse.kapua.locator.KapuaProvider;
+import org.eclipse.kapua.model.domain.Actions;
+import org.eclipse.kapua.model.id.KapuaId;
+import org.eclipse.kapua.model.query.KapuaQuery;
+import org.eclipse.kapua.service.authorization.AuthorizationService;
+import org.eclipse.kapua.service.authorization.permission.PermissionFactory;
+import org.eclipse.kapua.service.eventlog.EventLog;
+import org.eclipse.kapua.service.eventlog.EventLogCreator;
+import org.eclipse.kapua.service.eventlog.EventLogListResult;
+import org.eclipse.kapua.service.eventlog.EventLogService;
+import org.eclipse.kapua.service.user.User;
+import org.eclipse.kapua.service.user.UserService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link EventLogService} implementation.
+ */
+@KapuaProvider
+public class EventLogServiceImpl extends AbstractKapuaConfigurableService implements EventLogService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventLogServiceImpl.class);
+
+    private final KapuaLocator locator = KapuaLocator.getInstance();
+
+    private final AuthorizationService authorizationService = locator.getService(AuthorizationService.class);
+    private final PermissionFactory permissionFactory = locator.getFactory(PermissionFactory.class);
+    private final UserService userService = locator.getService(UserService.class);
+
+    /**
+     * Constructor
+     */
+    public EventLogServiceImpl() {
+        super(EventLogService.class.getName(), EVENT_LOG_DOMAIN, EventLogEntityManagerFactory.getInstance());
+    }
+
+    @Override
+    public EventLog create(EventLogCreator eventLogCreator) throws KapuaException {
+        //
+        // Argument Validation
```

Comment:
On argument validation please remove the `.get()` part form Id checks. 
We check on `.getScopeId` which, when `null`, is an invalid parameter.